src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/erasure-code-profile-form/erasure-code-profile-form-modal.component.html

```diff
@@ -104,14 +116,18 @@
             <span class="invalid-feedback"
                   *ngIf="form.showError('m', frm, 'min')"
                   i18n>Must be equal to or greater than 1.</span>
+            <span class="invalid-feedback"
+                  *ngIf="form.showError('m', frm, 'max')"
+                  i18n>Coding chunks have exceeded the available OSDs of {{deviceCount}}.</span>
```

Comment:
Might be just me, but its strange I get an error regarding a value that exceeds **0** , but if I put a **1** it works and starts failing in another place.

![image](https://user-images.githubusercontent.com/399326/80090392-f5712e00-854e-11ea-81fd-541f8e33c29d.png)
------
![image](https://user-images.githubusercontent.com/399326/80090368-edb18980-854e-11ea-844e-bab7c3e8941d.png)

Reply:
You don't have any OSDs? How did you do that? ;) May be I should warn the user that he can't create any functional rule without any OSD ;)

The error is showing only on the higher number so that you don't see the message on both fields, as it should be easier to lower down the higher number. It's also easier to lower m before k. That's what your are seeing here.

Behind the scenes the following condition runs `k+m <= available OSDs` if that fails an error is shown either on k or m.

Reply:
I have 3 OSDs, but it still shows that...
![image](https://user-images.githubusercontent.com/399326/80114951-b0abbe00-8573-11ea-91bb-85ea1190c79c.png)

Reply:
maybe I forgot to reload the backend... let me check

Reply:
after I create a new vstart, this no longer happens. 
but now I have the following:

1. When I open the modal, the default is **4** although max OSD is **3**. Should we automatically set it to the max possible value? 
![image](https://user-images.githubusercontent.com/399326/80116536-9ecb1a80-8575-11ea-9972-5ab78a7f3a64.png)
2. Still have errors if I set **k** to **3**, error message is misleading.
![image](https://user-images.githubusercontent.com/399326/80116712-d3d76d00-8575-11ea-8557-056b7c34b457.png)
3. **m** show error when I'm using **2**. From the description of this PR I think **k+m** should be lower or equel than **3** (OSDs), but the error message is misleading.
![image](https://user-images.githubusercontent.com/399326/80116814-f36e9580-8575-11ea-9b2a-ae5f5ac72de8.png)

Reply:
This are the default values wich would be used if you create an erasure coded profile without any arguments. The defaults are provided in order to not automatically set them if you would not set the field, which could produce a failure (as seen in your case). The failure is only seen if the profile is in use, with the new validation you can see if it fails before creating the profile an using it.

IMO we should use the defaults.